cmake/bob.cmake

```diff
@@ -10,7 +10,7 @@ function(bob_always_full_rpath)
   list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES
        "${CMAKE_INSTALL_PREFIX}/lib" isSystemDir)
   if("${isSystemDir}" STREQUAL "-1")
-    set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib" PARENT_SCOPE)
+    set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib" "${CMAKE_INSTALL_PREFIX}/lib64" PARENT_SCOPE)
```

Comment:
Adding the `lib64` here solved the issue. @cwsmith, is it fine? Or, is there any better way to do it?

Reply:
Interesting! With your commit (f1eae45f11a166b3f01e2dace233643711ec92be), I got `lib64` instead of `lib`.
```bash
Omega_h_ENABLE_GNUINSTALLDIRS="ON" ; BOB_LIB_DESTINATION="lib64" ; BOB_BIN_DESTINATION="bin"
```
Both ON and OFF for `-DBUILD_SHARED_LIBS` flag give `lib64`.
And the grep command returns nothing.

I am using `CMAKE_VERSION: 3.26.3`.

Reply:
Yar.  So, `bob_always_full_rpath()` is called here

https://github.com/SCOREC/omega_h/blob/f1eae45f11a166b3f01e2dace233643711ec92be/cmake/bob.cmake#L72

which is before `BOB_LIB_DESTINATION` is set.

Can you try moving the call to `bob_always_full_rpath()` down after `BOB_LIB_DESTINATION` is set and using the following to set `CMAKE_INSTALL_RPATH` in `bob_always_full_rpath()`?

```
set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${BOB_LIB_DESTINATION}" PARENT_SCOPE)
```